orbmain/src/main/java/com/sun/corba/ee/impl/io/IIOPInputStream.java

```diff
@@ -1152,6 +1153,32 @@ private synchronized Object inputObject(Class clz, ClassInfoCache.ClassInfo cinf
         return currentObject;
     }
 
+    // This is a horrible workaround for a problem that looks a bit challenging to fix now.
+    // In JDK11, the Date class was modified to call defaultWriteObject(), which doesn't actually do anything but
+    // set the "defaultWriteObjectCalled" flag in the serialization of the class; unfortunately, the
+    // deserialization logic cannot handle that case and throws an exception. We are therefore employing this
+    // workaround to allow Date objects to be read across the wire between JDK8 and JDK11.
+    // Fixing it will involve massive code simplification in order to eliminate duplicate code and allow for
+    // implementation of a simpler state machine that depends on the data sent rather than how it was produced.
+    private boolean readDefaultWriteObjectCalledFlag() throws IOException {
+        boolean sentDefaultWriteObjectCalled = readBoolean();
+
+        if (isDateClassWorkaroundRequired()) return getDateClassWorkaroundValue();
+        return sentDefaultWriteObjectCalled;
+    }
+
+    private boolean isDateClassWorkaroundRequired() {
+        return currentClassDesc.getName().equals(Date.class.getName());
+    }
+
+    private boolean getDateClassWorkaroundValue() {
+        return isJdk11_orLater();
+    }
+
+    private boolean isJdk11_orLater() {
+        return !System.getProperty("java.version").startsWith("1.");
```

Comment:
I assume this code also needs to work on JDK 8 so you can't use Runtime.Version.